test/tests/test_general.py

```diff
@@ -58,8 +58,23 @@ def test_pihole_ftl_clean_shutdown(docker):
     # Get the container ID from the docker fixture
     container_id = docker.backend.name
 
-    # Wait for startup
-    time.sleep(5)
+    # Wait for FTL to fully start up by checking logs
+    max_wait_time = 60  # Maximum wait time in seconds
```

Comment:
Thought (I might be wrong, I'm hungover and on a train):

In the other PR, we wait for 10 seconds for FTL to start - and if it does not, we stop the container. 

As the running code already has a built-in timeout, do we need to do this here too (and for longer...)?

Also maybe the test should be renamed to something like "test_pihole_ftl_starts_and_shuts_down_cleanly"